src/data/ability.ts

```diff
@@ -2262,14 +2262,15 @@ export class PostSummonTransformAbAttr extends PostSummonAbAttr {
       target = targets[0];
     }
 
+    target = target!; // compiler doesn't know its guranteed to be defined
     pokemon.summonData.speciesForm = target.getSpeciesForm();
     pokemon.summonData.fusionSpeciesForm = target.getFusionSpeciesForm();
     pokemon.summonData.ability = target.getAbility().id;
     pokemon.summonData.gender = target.getGender();
     pokemon.summonData.fusionGender = target.getFusionGender();
     pokemon.summonData.stats = [ pokemon.stats[Stat.HP] ].concat(target.stats.slice(1));
     pokemon.summonData.battleStats = target.summonData.battleStats.slice(0);
-    pokemon.summonData.moveset = target.getMoveset().map(m => new PokemonMove(m.moveId, m.ppUsed, m.ppUp));
+    pokemon.summonData.moveset = target.getMoveset().map(m => new PokemonMove(m!.moveId, m!.ppUsed, m!.ppUp)); // TODO: are those bangs correct?
```

Comment:
```suggestion
    pokemon.summonData.moveset = target.getMoveset().map(m => new PokemonMove(m?.moveId, m?.ppUsed, m?.ppUp));
```

I think in this case we should be checking for `null` based on `pokemon-data` moveset definition
```
public moveset: (PokemonMove | null)[];
```

Reply:
PokemonMove() constructor does not allow `undefined` for the move-id (ironic, isn't it?)